src/main/dynamo/operations/transactWriteItems.ts

```diff
@@ -33,5 +33,30 @@ export async function transactWriteItems<T extends TaggedModel>(params: Transact
     })
   )
 
-  await client.transactWrite({ TransactItems: requests }).promise()
+  const response = client.transactWrite({ TransactItems: requests })
+
+  // If a transaction is cancelled (i.e. fails), the AWS sdk sticks the reasons in the response
+  // body, but not the exception. So when errors occur, we extract the reasons (if present)
+  //  See: https://github.com/aws/aws-sdk-js/issues/2464#issuecomment-503524701
+  let transactionCancellationReasons: any[] | undefined
```

Comment:
Looked for a good way to write a unit test for this and so far I haven't been able to get the local version of DynamoDB to behave the same way as actual DynamoDB -- the local version doesn't appear to include the `CancellationReasons` in its results.